### PR TITLE
EXD-115 Form reset when modal is closed

### PR DESCRIPTION
**Pull request by KamilSilk**

### CODE REVIEW

File: src/components/CreateOrModifyTagDialog.tsx [hunk 1]
```diff
@@ -45,6 +45,12 @@ export default function CreateOrModifyTagDialog({
     },
   });
 
+  useEffect(() => {
```
Comment (MicDeb-Silk) [inline src/components/CreateOrModifyTagDialog.tsx]: Is it possible to use a callback on dialog close?

File: src/components/CreateOrModifyTagDialog.tsx [hunk 2]
```diff
@@ -1,132 +1,145 @@
...
 import {
-  Form,
-  FormControl,
-  FormField,
-  FormItem,
-  FormLabel,
-  FormMessage,
-} from "@/components/ui/form.tsx";
-import { zodResolver } from "@hookform/resolvers/zod";
+	Form,
```
Comment (MicDeb-Silk) [inline src/components/CreateOrModifyTagDialog.tsx:5]: Check prettier settings. Do you need double tab space?

File: src/components/CreateOrModifyTagDialog.tsx [hunk 3]
```diff
@@ -1,132 +1,145 @@
...
+		resolver: zodResolver(createOrModifyTagFormSchema),
+		defaultValues: {
+			name: '',
+		},
+	});
 
-  const { tagDetails, createTagHandler, updateTagHandler } =
-    useCreateOrModifyTag(currentTagId, () => setCurrentTagId(null), form.reset);
+	useEffect(() => {
+		if (modalState === 'ADD_TAG') {
```
Comment (MicDeb-Silk) [inline src/components/CreateOrModifyTagDialog.tsx]: If I understand correctly, you want to clear the form after closing the modal. You have an onOpenChange callback in the Dialog component. I think you can call form.rest() inside onOpenChange

File: src/components/CreateOrModifyTagDialog.tsx [hunk 4]
```diff
@@ -1,132 +1,145 @@
-import { Input } from "@/components/ui/input.tsx";
-import { Button } from "@/components/ui/button.tsx";
-import { useForm } from "react-hook-form";
+import { Input } from '@/components/ui/input.tsx';
```
Comment (MicDeb-Silk) [inline src/components/CreateOrModifyTagDialog.tsx:1]: In addition to prettier, add also better eslint rules.
1. Better import order
2. Import files without extensions

File: src/components/CreateOrModifyTagDialog.tsx [hunk 5]
```diff
@@ -1,132 +1,145 @@
...
+													/>
+												</FormControl>
+											</div>
+										</div>
+										<FormMessage />
+									</FormItem>
+								)}
+							/>
+							<DialogFooter>
+								<Button type="submit">Save tag</Button>
```
Comment (MicDeb-Silk) [inline src/components/CreateOrModifyTagDialog.tsx:140]: You can add dictionary to the project

File: src/components/TagsManager.tsx [hunk 6]
```diff
@@ -2,21 +2,23 @@ import { Toaster } from "@/components/ui/toaster.tsx";
 import ContentWrapper from "@/components/ContentWrapper.tsx";
 import { TagsContent } from "@/components/TagsContent.tsx";
 import { Tabs, TabsList, TabsTrigger } from "@/components/ui/tabs.tsx";
-import { useState } from "react";
 import PrimaryActionButton from "@/components/buttons/PrimaryActionButton.tsx";
 import { PlusCircle } from "lucide-react";
 import CreateOrModifyTagDialog from "@/components/CreateOrModifyTagDialog.tsx";
+import {useModalStore} from '@/store/modalStore.ts';
```
Comment (MicDeb-Silk) [inline src/components/TagsManager.tsx:8]: also add eslint rule for better object forrmating

File: src/store/modalStore.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+export type ModalStateKeys = keyof typeof MODAL_STATE;
+
+const initialState = {
+	isModalOpen: false,
+	selectedId: null,
+	modalState: undefined,
+	modalProps: undefined,
+};
+
+export type ModalPayload =
```
Comment (MicDeb-Silk) [inline src/store/modalStore.ts:32]: Why do you creat union with whole ModalPaylod object insted 
```
{
	modalState: typeof MODAL_STATE.ADD_TAG | typeof MODAL_STATE.EDIT_TAG |  MODAL_STATE.REMOVE_TAG;
	params: ModalProps;
}
```
alternatively, will ModalProps change for different modalState?

File: src/components/CreateOrModifyTagDialog.tsx (hunk 3 above)
Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx]: In this case I want to clear form state in the edit modal, only when user close this modal. Add modal should be empty after user open it, When he writes data inside, and accidently close it, the data should be there when he open the modal again.

Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx]: I will remove this useEffect and make a condition inside onOpenChange prop to reset form when modalState is "EDIT_TAG" just like you write above.

File: src/components/CreateOrModifyTagDialog.tsx (hunk 4 above)
Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx:1]: There is another task (117) for the eslint rules.

File: src/components/TagsManager.tsx (hunk 6 above)
Comment (KamilSilk) [reply src/components/TagsManager.tsx:8]: There is another task (117) for the eslint rules.

File: src/components/CreateOrModifyTagDialog.tsx (hunk 5 above)
Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx:140]: The dictionary also will be added in another task

File: src/store/modalStore.ts (hunk 7 above)
Comment (KamilSilk) [reply src/store/modalStore.ts:32]: As i saw in the application - no. For all modals the required field is idNumber to get the data from api.

File: src/components/CreateOrModifyTagDialog.tsx (hunk 2 above)
Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx:5]: There is another task (117) for the eslint rules.

File: src/components/CreateOrModifyTagDialog.tsx (hunk 1 above)
Comment (KamilSilk) [reply src/components/CreateOrModifyTagDialog.tsx]: This useEffect is to remove after the onOpenChange solution

**Pull request merged**